### DISCUSSION TITLE
Forward declaration for fkyaml::node

### DISCUSSION THREAD

**Discussion by abcminiuser** [fktn-k/fkYAML#420] [Ideas]
Hello - thanks for the library!

It would be great if the standard `fkyaml::node` could be forward declared in a separate header, so that translation units that don't directly use the library can be built without having to pay for the compile time cost of including the full library header. Ideally this means I could use the forward-declaration version in my file loader class header, and then only include the actual library code inside the one CPP file that needs to work with the nodes.

**Comment by fktn-k** [comment]
@abcminiuser   
Thanks for sharing your idea!  
I agree with this:  
> Ideally this means I could use the forward-declaration version in my file loader class header, and then only include the actual library code inside the one CPP file that needs to work with the nodes.

Regarding such a header file which contains forward declarations, I'm planning to add one which is similar to the ones in [nlohmann json](https://github.com/nlohmann/json/blob/develop/include%2Fnlohmann%2Fjson_fwd.hpp) or in [spdlog](https://github.com/gabime/spdlog/blob/v1.x/include%2Fspdlog%2Ffwd.h). They contains forward declarations of types which either implement library APIs or provide a customization point. Is that what you expect?

**Reply by abcminiuser** [reply]
Yes, the `<nlohmann/json_fwd.hpp>` is actually the header I was thinking of when I suggested this. Having an equivalent here would be great.

**Reply by fktn-k** [reply]
I got your point. Thanks!

**Reply by fktn-k** [reply]
@abcminiuser   
The header file for forward declarations has been added in the PR #422.  
You can use the file `fkyaml_fwd.hpp` from either the `include` or `single_include` directory for forward declarations.
